test/mtx-test.js

```diff
@@ -62,4 +70,600 @@ describe('MTX', function() {
     assert.ok(view instanceof CoinView);
     assert.deepStrictEqual(got, want);
   });
+
+  describe('Fund', function() {
+    const wallet1 = new MemWallet();
+    const wallet2 = new MemWallet();
+
+    const coins1 = [
+      dummyCoin(wallet1.getAddress(), 1000000),
+      dummyCoin(wallet1.getAddress(), 1000000),
+      dummyCoin(wallet1.getAddress(), 1000000),
+      dummyCoin(wallet1.getAddress(), 1000000),
+      dummyCoin(wallet1.getAddress(), 1000000)
+    ];
+
+    const last1 = coins1[coins1.length - 1];
+    const last2 = coins1[coins1.length - 2];
+
+    /**
+     * Test matrix
+     * fund w/o inputs, just coins
+     * fund with preferred inputs - no view && coins
+     * fund with preferred inputs - view    && no coins
+     * fund with preferred inputs - view    && coins
+     * fund with preferred inputs - no view && coins - error
+     *
+     * fund with existing inputs  - no view && coins
+     * fund with existing inputs  - view    && no coins
+     * fund with existing inputs  - view    && coins
+     * fund with existing inputs  - no view && no coins - error
+     *
+     * fund with both inputs - no view && coins(1e, 1p)
+     * fund with both inputs - view(1e, 1p) && no coins
+     * fund with both inputs - view(1e, 1p) && coins(1e, 1p)
+     * fund with both inputs (1e, 1p) - no view(1e) && no coins(1e) - error.
+     * fund with both inputs (1e, 1p) - no view(1p) && no coins(1p) - error.
+     * fund with both inputs (1e, 1p) - no view && no coins - error.
+     */
+
+    it('should fund mtx', async () => {
+      const mtx = new MTX();
+
+      mtx.addOutput(wallet2.getAddress(), 1500000);
+
+      await mtx.fund(coins1, {
+        changeAddress: wallet1.getChange()
+      });
+
+      assert.strictEqual(mtx.inputs.length, 2);
+      assert.strictEqual(mtx.outputs.length, 2);
+    });
+
+    it('should fund with preferred inputs - coins', async () => {
+      const mtx = new MTX();
+      const coin = last1;
+
+      mtx.addOutput(wallet2.getAddress(), 1500000);
+
+      await mtx.fund(coins1, {
+        changeAddress: wallet1.getChange(),
+        inputs: [{
+          hash: coin.hash,
+          index: coin.index
+        }]
+      });
+
+      assert.strictEqual(mtx.inputs.length, 2);
+      assert.strictEqual(mtx.outputs.length, 2);
+
+      assert.bufferEqual(mtx.inputs[0].prevout.hash, coin.hash);
+      assert.strictEqual(mtx.inputs[0].prevout.index, coin.index);
+
+      assert(mtx.view.hasEntry({
+        hash: coin.hash,
+        index: coin.index
+      }));
+    });
+
+    it('should fund with preferred inputs - view', async () => {
+      const mtx = new MTX();
+      const coin = dummyCoin(wallet1.getAddress(), 1000000);
+
+      mtx.addOutput(wallet2.getAddress(), 1500000);
+      mtx.view.addCoin(coin);
```

Comment:
Could also be `mtx.addCoin(coin)` right? I tried this and test passed -- is there any significant difference?

Reply:
Same as above. We want to make sure view is processed properly.